File: CampusView/backend/src/main/java/com/example/campusview/entity/ScenicSpot.java
package com.example.campusview.entity;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "scenic_spots")
@Data
@NoArgsConstructor
public class ScenicSpot {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    @Column(columnDefinition = "TEXT")
    private String description;

    private String location;

    private String imagePath;

    private String coordinates;

    @Column(columnDefinition = "TEXT")
    private String features;

    public ScenicSpot(String name, String description) {
        this.name = name;
        this.description = description;
    }
} 

File: CampusView/app/src/main/java/com/example/campusview/data/entity/ScenicSpot.kt
package com.example.campusview.data.entity

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "scenic_spots")
data class ScenicSpot(
    @PrimaryKey(autoGenerate = true)
    val id: Long = 0,
    val name: String,
    val description: String?,
    val location: String?,
    val imagePath: String?,
    val coordinates: String?, // 经纬度坐标
    val features: String? // 用于图像匹配的特征数据
) 